librosa/display.py

```diff
@@ -917,5 +969,10 @@ def __coord_n(n, **_kwargs):
 
 
 def __coord_time(n, sr=22050, hop_length=512, **_kwargs):
-    '''Get time coordinates from frames'''
+    '''Get time coordinates in seconds from frames'''
     return core.frames_to_time(np.arange(n+1), sr=sr, hop_length=hop_length)
+
+
+def __coord_milliseconds(n, sr=22050, hop_length=512, **_kwargs):
+    '''Get time coordinates in milliseconds from frames'''
+    return 1000 * __coord_time(n, sr, hop_length, **_kwargs)
```

Comment:
I think this is the problem with your mis-aligned plots.

All time-like plots should use `__coord_time` and have positions in seconds; `__coord_milliseconds` should not exist.  All that we need to do is make sure that `__decorate_axis` sets the timeformatter correctly, which it seems to.

Reply:
Ah..I see! 
But, when we allow unit specification to `ms` in specshow(), doesn't this mean that we need unit conversion from seconds to milliseconds? Currently, this is not handled in `__decorate_axis`.

Reply:
Yes, but that should only be done within the TimeFormatter (and when specifying the axis label).  The coordinate grid should still be in seconds.

Reply:
Oops forgot about not allowing time unit specification in waveplot.
Adding unit conversion in timeformatter works and fixes the bug :)

One last thing though.. the axis label will conflict with the actual plotted unit, since currently the axis labeling is hard coded within `__decorate_axis`, it does not take into account the `sharex` situation. Hmm...
<img src="https://user-images.githubusercontent.com/19829532/44438056-5e750580-a5f8-11e8-9963-a343b8cd5b72.png" width="400">